### PR TITLE
Fix not to create unnecessary `thread_params`

### PR DESCRIPTION
**Pull request by HJLeee**
In case of `EP_THREAD_TYPE_SERVER`, thread_param is a leak in the original code.

### CODE REVIEW

**Comment by HJLeee** [comment]
- runtime (coreclr Pri0 Runtime Tests Run windows x86 checked) 
https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-76007-merge-ac12cc893b1f4aa0bc/baseservices.finalization/1/console.34afcb43.log?helixlogtype=result
This might be related to https://github.com/dotnet/runtime/pull/75952?

**Comment by HJLeee** [comment]
#76046

**Comment by HJLeee** [comment]
/azp run runtime
to run runtime (Installer Build and Test coreclr OSX_x64 Release) Cancelled after 124m — Installer Build and Test coreclr OSX_x64 Release was canceled

**Comment by danmoseley** [comment]
@HJLeee unfortunately the azp commands do not work without write access. You can rerun checks by closing and reopening or pushing an empty commit. Of course if you just want to rerun one pipeline it’s more efficient to do that but at the moment you have to ask someone with commit access to do it 😕

**Comment by HJLeee** [comment]
Thank you for the tip. Seems useful to the commiters if timeout tests can be run again. :)

**Comment by HJLeee** [comment]
@jkotas Please take a look.

File: src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h [hunk 1]
```diff
@@ -2012,27 +2012,39 @@ ep_rt_thread_create (
...
-				if (thread_params->thread_params.thread->CreateNewThread (0, ep_rt_thread_coreclr_start_func, thread_params)) {
+
+				if (thread_params->thread_params.thread->CreateNewThread (0, ep_rt_thread_coreclr_start_func, thread_params))
+				{
 					thread_params->thread_params.thread->SetBackground (TRUE);
 					thread_params->thread_params.thread->StartThread ();
+
 					if (id)
+					{
 						*reinterpret_cast<DWORD *>(id) = thread_params->thread_params.thread->GetThreadId ();
```
Comment (jkotas) [inline src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h:2046]: `thread_params` can be freed by the newly started thread by this point. The thread id should be captured e.g. before the thread is started.

File: src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h [hunk 2]
```diff
@@ -2012,27 +2012,39 @@ ep_rt_thread_create (
...
+				{
+					*reinterpret_cast<DWORD *>(id) = thread_id;
+				}
+				::CloseHandle (server_thread);
+				result = true;
+			}
+		}
+		else if (thread_type == EP_THREAD_TYPE_SESSION || thread_type == EP_THREAD_TYPE_SAMPLING)
+		{
+			rt_coreclr_thread_params_internal_t *thread_params = new (nothrow) rt_coreclr_thread_params_internal_t ();
```
Comment (jkotas) [inline src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h:2031]: `thread_params` is still going to leak when the thread fails to start (`CreateNewThread` fails).

File: src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h (hunk 1 above)
Comment (HJLeee) [reply src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h:2043]: Thank you. Fixed.

File: src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h (hunk 2 above)
Comment (HJLeee) [reply src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h:2031]: Thank you. Fixed.

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**